app/services/puppet_fact_parser.rb

```diff
@@ -145,6 +145,29 @@ def boot_timestamp
     uptime_seconds.nil? ? nil : (Time.zone.now.to_i - uptime_seconds.to_i)
   end
 
+  def virtual
+    facts['is_virtual']
+  end
+
+  def ram
+    facts['memorysize']
+    if facts.key?('memory')
+      value = facts.fetch('memory', {}).fetch('system', {}).fetch('total_bytes', nil)
```

Comment:
I think this would also work:
```ruby
if (value = facts.dig('memory', 'system', 'total_bytes'))
  value / 1.megabyte
else
  facts['memorysize_mb']
end
```